packages/core/src/tools/instrumentMethod.ts

```diff
@@ -125,11 +126,14 @@ export function instrumentSetter<TARGET extends { [key: string]: any }, PROPERTY
     return { stop: noop }
   }
 
+  const timeoutIds = new Set<TimeoutId>()
   let instrumentation = (target: TARGET, value: TARGET[PROPERTY]) => {
     // put hooked setter into event loop to avoid of set latency
-    setTimeout(() => {
+    const timeoutId = setTimeout(() => {
+      timeoutIds.delete(timeoutId)
       after(target, value)
     }, 0)
+    timeoutIds.add(timeoutId)
   }
```

Comment:
**💬 suggestion:** ‏This function is performance sensitive. To avoid manipulating a collection of timeouts, ou could have something like:

```ts
  let stopped = false
  let instrumentation = (target: TARGET, value: TARGET[PROPERTY]) => {
    // put hooked setter into event loop to avoid of set latency
    setTimeout(() => {
      if (!stopped) {
        after(target, value)
      }
    }, 0)
  }
  // ...
  stop() {
    // ...
    stopped = true
  }